src/components/DecisioningEngine/createEventRegistry.js

```diff
@@ -132,5 +139,16 @@ export default ({ storage }) => {
     return events[eventType][eventId];
   };
 
-  return { addExperienceEdgeEvent, addEvent, getEvent, toJSON: () => events };
+  const setStorage = newStorage => {
```

Comment:
I think it's redundant to have both `updateStorage` and `setStorage` functions.

I recommend removing this version of `setStorage` and renaming the `updateStorage` method above to `setStorage`.  Implementation of `setStorage` can be modified to accept `newStorage`  for example...

```js
  let currentStorage;
  let restore;
  let save;
  let events;

  const setStorage = newStorage => {
    currentStorage = newStorage;

    restore = createRestoreStorage(currentStorage, STORAGE_KEY);
    save = createSaveStorage(
      currentStorage,
      STORAGE_KEY,
      createEventPruner(MAX_EVENT_RECORDS, RETENTION_PERIOD)
    );
    events = restore({});
  };

  setStorage(storage);
```